Documentation/ApiOverview/Events/EventDispatcher/Index.rst

```diff
@@ -312,21 +346,32 @@ Existing event listeners can be overridden by custom implementations. For exampl
 If you want to replace this event listener with your custom implementation, your extension can
 achieve this by specifying:
 
+..  literalinclude:: _ServicesOverrideCustom.php
+    :language: yaml
+    :caption: EXT:my_extension/Configuration/Classes/EventListener/MySeoEvent.php
+
+or via :file:`Services.yaml` legacy declaration:
+
 ..  literalinclude:: _ServicesOverrideCustom.yaml
     :language: yaml
     :caption: EXT:my_extension/Configuration/Services.yaml
 
-Make sure that you set the :yaml:`identifier` attribute to exactly the string that the
-original implementation uses. If the identifier is not mentioned specifically in the original
-implementation, the fully-qualified name of the event listener class is used. So if
-:yaml:`identifier: 'ext-some-extension/modify-hreflang'` would not be there, the identifier would be set
-to :yaml:`identifier: 'SomeVendor\SomeExtension\Seo\HrefLangEvent'` and you would need to use that
+Make sure that you set the `identifier` property (either in YAML or the PHP attribute) to exactly
+the string that the original implementation uses. If the identifier is not mentioned specifically
+in the original implementation, the service name (usually the fully-qualified name of the event
```

Comment:
and unusually? When not defined, it is always the FQCN - or do I miss something?

Reply:
Inside services.yaml you are free what you use as a service name, it must not be a FQCN, it can be any unique string. FQCNs are recommended, but not required. So it falls back to a SERVICE NAME, not a FQCN actually. That's why I used "usually the FQCN", because sane people should use that, but it can't be ensure :D